src/compiler/utilities.ts

```diff
@@ -3871,7 +3872,7 @@ namespace ts {
             // Otherwise, keep printing a literal \0 for the null character
             return "\\0";
         }
-        return escapedCharsMap.get(c) || encodeUtf16EscapeSequence(c.charCodeAt(0));
+        return escapedCharsMap.get(c) ?? encodeUtf16EscapeSequence(c.charCodeAt(0));

```

Comment:
`||` should have been fine, since none of the entries are the empty string.
```suggestion
        return escapedCharsMap.get(c) || encodeUtf16EscapeSequence(c.charCodeAt(0));
```

Reply:
I thought that it's clearer to uses `??` than to rely on the implicit fact that the map doesn't include an empty string.  So to clarify -- are you saying that:
* You prefer a `||`?  (maybe because the downleveled check can be slow?)
* or is this a general preference for not changing stuff that works?

(In any case, I changed it back.)

Reply:
It was a request some time back from @DanielRosenwasser that we continue to use `||` unless `??` is absolutely necessary due to the emit complexity of `??` (increased size, decreased readability when looking at the JS).